extensions/arc/deployment/src/main/java/io/quarkus/arc/deployment/ConfigBuildStep.java

```diff
@@ -103,6 +102,41 @@ void analyzeConfigPropertyInjectionPoints(BeanDiscoveryFinishedBuildItem beanDis
                 continue;
             }
 
+            AnnotationInstance configProperty = injectionPoint.getRequiredQualifier(MP_CONFIG_PROPERTY_NAME);
+            if (configProperty != null) {
+                // Register a custom bean for injection points that are not handled by ConfigProducer
+                Type injectedType = injectionPoint.getType();
+                if (!isHandledByProducers(injectedType)) {
+                    customBeanTypes.add(injectedType);
+                }
+            }
+        }
+
+        for (Type type : customBeanTypes) {
+            if (type.kind() != Kind.ARRAY) {
+                // Implicit converters are most likely used
+                reflectiveClass.produce(new ReflectiveClassBuildItem(true, false, type.name().toString()));
+            }
+            DotName implClazz = type.kind() == Kind.ARRAY ? DotName.createSimple(ConfigBeanCreator.class.getName())
+                    : type.name();
+            syntheticBeans.produce(SyntheticBeanBuildItem.configure(implClazz)
+                    .creator(ConfigBeanCreator.class)
+                    .providerType(type)
+                    .types(type)
+                    .addQualifier(MP_CONFIG_PROPERTY_NAME)
+                    .param("requiredType", type.name().toString()).done());
+        }
+    }
+
+    @BuildStep
+    void validateConfigProperties(ValidationPhaseBuildItem validationPhase,
```

Comment:
The method name is a bit misleading and there's already a method called `validateConfigProperties`.

Reply:
Sure, let me find a better name :)